index/scorch/scorch.go

```diff
@@ -658,6 +658,16 @@ func (s *Scorch) StatsMap() map[string]interface{} {
 	m["num_persister_nap_merger_break"] = m["TotPersisterMergerNapBreak"]
 	m["total_compaction_written_bytes"] = m["TotFileMergeWrittenBytes"]
 
+	// the bool stat `index_active` indicates whether the background routines
+	// which causes the index to reach a steady state are still doing some work.
+	if rootEpoch, ok := m["CurRootEpoch"].(uint64); ok {
+		if lastMergedEpoch, ok := m["LastMergedEpoch"].(uint64); ok {
+			if lastPersistedEpoch, ok := m["LastPersistedEpoch"].(uint64); ok {
+				m["index_active"] = !(lastMergedEpoch == rootEpoch && lastPersistedEpoch == rootEpoch)
```

Comment:
Separately, wouldn't this stat incorrectly show as active in the situation should the merge policy prevent the merger from running over the last batch of items?

Reply:
in the case where there were no merged segments introduced, the LastMergedEpoch would still get updated with the latest root epoch's value. Assuming that the merger doesn't do any active merging of segments for a while (based on merge policy) it would remain equal to the root epoch and the stat value over here would be false (indicating that the index is not doing any activity in the background)

Reply:
Ack.